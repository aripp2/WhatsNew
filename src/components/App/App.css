* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: white;
}

.app {
  display: flex;
  background: #141527;
  text-align: center;
  width: 100%;
}

main {
  width: 80%;
  display: flex;
  flex-direction: column;
}
