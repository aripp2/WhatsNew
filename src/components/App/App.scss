@import './src/variables.scss';

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: $highlight-color;
}

.app {
  display: flex;
  background: $dark-color;
  text-align: center;
  width: 100%;
}

main {
  width: 80%;
  display: flex;
  flex-direction: column;
}


