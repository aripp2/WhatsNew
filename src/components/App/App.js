import React, { Component } from 'react';
import local from '../../data/local';
import Menu from '../Menu/Menu';
import SearchForm from '../SearchForm/SearchForm';
import NewsContainer from '../NewsContainer/NewsContainer';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      local
    }
  }

  render () {
    return (
      <div className="app">
        <Menu />
        <SearchForm />
        <NewsContainer news="this.state"/>
      </div>
    );
  }
}

export default App;
