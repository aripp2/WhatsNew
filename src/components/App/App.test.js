import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  let wrapper;
  const mockChangeNewsType = jest.fn();
  
  const mockNews = {
    mockLocal: [{
      id: 1,
      headline: 'Local Headline One',
      img: 'Local Image One',
      description: 'Local Description One',
      url: 'Local URL One'
    },
    {
      id: 2,
      headline: 'Local Headline Two',
      img: 'Local Image Two',
      description: 'Local Description Two',
      url: 'Local URL Two'
    }],
    mockHealth: [{
      id: 1,
      headline: 'Health Headline One',
      img: 'Health Image One',
      description: 'Health Description One',
      url: 'Health URL One'
    },
    {
      id: 2,
      headline: 'Health Headline Two',
      img: 'Health Image Two',
      description: 'Health Description Two',
      url: 'Health URL Two'
    }]
  }

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when changeNewsType is called', () => {
    wrapper.instance().changeNewsType('mockHealth');
    console.log(wrapper.state().currentNews)

    expect(wrapper.state('currentNews')).toEqual(mockNews.mockHealth);

  });

});