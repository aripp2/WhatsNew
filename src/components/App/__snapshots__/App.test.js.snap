// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="app"
>
  <Menu
    changeNewsType={[Function]}
  />
  <main>
    <SearchForm
      searchNews={[Function]}
    />
    <NewsContainer
      news={Array []}
    />
  </main>
</div>
`;
